### PR TITLE
Update api_changes_list_2024.md

### PR DESCRIPTION
**Pull request by jreznot**
Document breaking changes in API of com.intellij.util.CachedValueBase

### CODE REVIEW

File: reference_guide/api_changes_list_2024.md [hunk 1]
```diff
@@ -147,6 +147,14 @@ Class `com.intellij.diff.editor.DiffVirtualFile` now extends `com.intellij.diff.
 `com.intellij.openapi.util.io.NioPathUtil.isAncestor(Path, Path, boolean)` method removed
 : Use `Path.startsWith()` instead.
 
+`com.intellij.util.CachedValueBase.setData(CachedValueBase.Data)` abstract method added
+: Must be implemented
```
Comment (YannCebron) [inline reference_guide/api_changes_list_2024.md:151]: description should end with `.`

File: reference_guide/api_changes_list_2024.md [hunk 2]
```diff
@@ -147,6 +147,14 @@ Class `com.intellij.diff.editor.DiffVirtualFile` now extends `com.intellij.diff.
...
+
+`com.intellij.util.CachedValueBase.getRawData()` abstract method added
+: Must be implemented
```
Comment (YannCebron) [inline reference_guide/api_changes_list_2024.md:154]: description should end with `.`

File: reference_guide/api_changes_list_2024.md [hunk 3]
```diff
@@ -147,6 +147,14 @@ Class `com.intellij.diff.editor.DiffVirtualFile` now extends `com.intellij.diff.
...
+
+Class `com.intellij.util.CachedValuesFactory` no longer part of the API
+
```
Comment (YannCebron) [inline reference_guide/api_changes_list_2024.md:157]: description/solution is mandatory

**Review by YannCebron — CHANGES_REQUESTED** [review]

File: reference_guide/api_changes_list_2024.md (hunk 3 above)
Comment (jreznot) [reply reference_guide/api_changes_list_2024.md:157]: There is no solution available, should I write this directly?

File: reference_guide/api_changes_list_2024.md (hunk 2 above)
Comment (jreznot) [reply reference_guide/api_changes_list_2024.md:154]: Would be great to have some tooling to detect that in IDE

File: reference_guide/api_changes_list_2024.md (hunk 3 above)
Comment (YannCebron) [reply reference_guide/api_changes_list_2024.md:157]: If this is the best answer, yes.

**Pull request merged**